pkg/assert/validate.go

```diff
@@ -97,3 +118,43 @@ func ValidateOnNode(ip string, args ...string) error {
 	}
 	return validate(exec, args...)
 }
+
+func addResult(command, assertion, result string) {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	results = append(results, TestResult{Command: "\nCommand:\n" + command + "\n", Assertion: "\nAssertion:\n" +
+		assertion + "\n", Result: "\nMatched with result:\n" + result + "\n"})
+}
+
+// PrintResults prints the results from the tests called
+func PrintResults() {
```

Comment:
Can we update the tests themselves so that they print their own results? This feels like an arbitrary call that is prone to miss needed changes when changes are made to those tests.

Reply:
tbh i dont want to add anything like that in any test, i want to have the flexibilty to only call where i want when i want , if i understood correctly your request.

so in this way, we only call this gather and print of results, in where we want.